src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2369,7 +2369,10 @@ def _reindex_axis_0(
         """
         self._raise_not_implemented_error_for_timedelta()
 
-        new_index_qc = pd.Series(labels)._query_compiler
+        if isinstance(labels, pd.Index):
```

Comment:
```suggestion
        if isinstance(labels, (pd.Index, pandas.Index)):
```
I believe if we do this, we can fix the name bug too.

Reply:
I don't think that would work directly since native pandas Series don't have a query compiler attribute

Reply:
Then you convert it to pd.Index first.